docs/builtins.md

```diff
@@ -0,0 +1,730 @@
+A, second: B`.
+
+Full list of valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| %v        | the argument in a default format  |
+| %%        | a literal percent sign `%`; consumes no value  |
+| %t        | the word `true` or `false` |
+| %f        | decimal point (floating-point) number, e.g. 123.456000 |
+| %e        | scientific notation, e.g. -1.234456e+78 |
+| %s        | string value |
+| %q        | a double-quoted string |
+
+If the arguments for the `%s`, `%q`, `%f`, `%e`, and `%t` specifiers do
+not match the required type, a runtime error will occur.
+
+The _width_ and _precision_ of a floating-point number can be specified
+with the `%f` and `%v` format specifiers.
+
+- Width is the number of characters that will be used to print the
+  number. If the width is not specified, it will be calculated based on
+  the size of the number. It can be useful for padding and aligned
+  output.
+- Precision is the number of decimal places that will be displayed. If
+  the precision is not specified, it will be set to 6. Here is a table
+  that shows the different ways to specify the width and precision of a
+  floating-point number:
+
+
+| Verb    | Description |
+| ------- | ------------- |
+| %f      | default width, default precision |
+| %7f     | width 7, default precision |
+| %.2f    | default width, precision 2 |
+| %7.2f   | width 7, precision 2 |
+| %7.f    | width 7, precision 0 |
+
+
+The `-` flag tells `printf` to pad the output with spaces on the right,
+rather than the left. The `0` flag tells `printf` to pad the output
+with leading zeros, rather than spaces.
+
+The width number and the align-left flag  `-` can be used with all valid
+specifiers. For example:
+
+```evy
+printf "right:  |%7.2f|\n" 1
+printf "left:   |%-7.2f|\n" 1
+printf "zeropad:|%07.2f|\n" 1
+```
+Output
+```
+1 2.000000 3
+right:  |   1.00|
+left:   |1.00   |
+zeropad:|0001.00|
+```
+
+## Types
+
+### `len`
+
+`len` returns the number of characters in a string, the number of
+elements in an array or the number of key-value pairs in a map.
+
+#### Example
+
+```evy
+l := len "abcd"
+print "len \"abcd\":" l
+
+l = len [1 2]
+print "len [1 2]:" l
+
+l = len {a:3 b:4 c:5}
+print "len {a:3 b:4 c:5}:" l
+```
+
+Output
+```
+len "abcd": 4
+len [1 2]: 2
+len {a:3 b:4 c:5}: 3
+```
+
+#### Reference 
+
+    len:num a:any
+
+The `len` function takes a single argument, which can be a string, an
+array, or a map. If the argument is a string, `len` returns the number
+of characters in the string. If the argument is an array, `len` returns
+the number of elements in the array. If the argument is a map, `len`
+returns the number of key-value pairs in the map. If the argument is of
+any other type, `len` will cause a runtime error.
+
+### `typeof`
+
+`typeof` returns the type of the argument as string value.
+
+#### Example
+
+```evy
+a:any
+a = "abcd"
+t := typeof a
+print "typof \"abcd\":" t
```

Comment:
s/typof/typeof/ - and in the output section below too.

Reply:
done.